docs/data/toolpad/core/components/sign-in-page/sign-in-page.md

```diff
@@ -44,15 +44,86 @@ The following providers are supported and maintained by default:
 - Twitch
 - Discord
 - Keycloak
-- Credentials (username/password)
 
 Find details on how to set up each provider in the [Auth.js documentation](https://authjs.dev/getting-started/authentication/oauth).
 :::
 
+## Magic Link
+
+You can use the `SignInPage` component to quickly set up authentication via one-time verification links. It uses Nodemailer under the hood to send the verification link to the user's email address. See more details in the [Auth.js docs](https://authjs.dev/getting-started/providers/nodemailer/) on configuration and customization.
+
+To render a magic link form, pass in a provider with `nodemailer` as the `id` property.
+
+{{"demo": "MagicLinkSignInPage.js", "iframe": true, "height": 500}}
+
+### Alerts
+
+The `SignInPage` component can display a success alert if the email is sent successfully. You can enable this by passing a `success` property in the
+response object of the `signIn` prop.
+
+{{"demo": "MagicLinkAlertSignInPage.js", "iframe": true, "height": 500}}
+
+To get the magic link working, you need to add the following code to your custom sign-in page:
+
+```tsx title="app/auth/signin/page.tsx"
+import * as React from 'react';
+import { SignInPage } from '@toolpad/core/SignInPage';
+import { AuthError } from 'next-auth';
+import type { AuthProvider } from '@toolpad/core';
+import { signIn, providerMap } from '../../../auth';
+
+export default function SignIn() {
+  return (
+    <React.Fragment>
+      <SignInPage providers={providerMap} signIn={
+        async (provider: AuthProvider, formData: FormData, callbackUrl?: string) {
+          try {
+            return await signInAction(provider.id, {
+              ...(formData && {
+                email: formData.get('email'),
+                password: formData.get('password'),
+              }),
+              redirectTo: callbackUrl ?? '/',
+            });
+        } catch (error) {
+          if (error instanceof Error && error.message === 'NEXT_REDIRECT') {
+            if (provider.id === 'nodemailer' &&
+            (error as any).digest?.includes('verify-request')) {
+            return {
+              success: 'Check your email for a verification link.',
```

Comment:
Would a more specific name like `successAlert` or `successMessage` be better?

Reply:
Got it, just a thought, I think `error` shouldn't be renamed so we can keep `success` if it's more consistent with that.